### PR TITLE
dns/bind: check&preview primary zone, log to syslog with desired severity

### PR DESCRIPTION
**Pull request by kulikov-a**
Hi!
in continuation of https://github.com/opnsense/plugins/pull/3704
- add 'check & preview' button to Primary Zone grid for better understand\debugging zones config
- move general log to syslog (and register app and facility for remote logging)
- make grids responsive

Thanks! 

https://github.com/opnsense/plugins/assets/36099472/262c770f-741f-49f2-8bd9-04c3ee6a569d

### CODE REVIEW

**Review by mimugmail — APPROVED** [review]
Looks really nice, regarding coding style please @fichtner have a final look

**Comment by kulikov-a** [comment]
@mimugmail thanks!
ver bump added

File: dns/bind/src/etc/inc/plugins.inc.d/bind.inc [hunk 1]
```diff
@@ -99,3 +99,18 @@ function bind_configure_do($verbose)
...
+ *  register syslog facilities
+ * @return array
+ */
+function bind_syslog()
+{
+    $syslogconf = array();
+
+    $syslogconf['bind'] = array(
+        'facility' => array('named'),
+    );
```
Comment (fichtner) [inline dns/bind/src/etc/inc/plugins.inc.d/bind.inc:113]:
```suggestion
    $syslogconf = [];

    $syslogconf['bind'] = ['facility' => ['named']];
```

File: dns/bind/src/opnsense/mvc/app/controllers/OPNsense/Bind/Api/GeneralController.php [hunk 2]
```diff
@@ -31,9 +31,26 @@
...
+use OPNsense\Core\Backend;
 
 class GeneralController extends ApiMutableModelControllerBase
 {
     protected static $internalModelClass = '\OPNsense\Bind\General';
     protected static $internalModelName = 'general';
+
+    public function zonetestAction($zonename = null)
+    {
+        $zonename = $_POST['zone'];
```
Comment (fichtner) [inline dns/bind/src/opnsense/mvc/app/controllers/OPNsense/Bind/Api/GeneralController.php:43]: Using $_POST is a bit unusual for Phalcon

File: dns/bind/src/opnsense/mvc/app/controllers/OPNsense/Bind/Api/GeneralController.php [hunk 3]
```diff
@@ -31,9 +31,26 @@
...
+        $backend = new Backend();
+        $response = trim($backend->configdRun("bind check_zone " . $zonename));
+        return array("response" => $response);
+    }
+    
+    public function zoneshowAction($zonename = null)
+    {
+        $zonename = $_POST['zone'];
```
Comment (fichtner) [inline dns/bind/src/opnsense/mvc/app/controllers/OPNsense/Bind/Api/GeneralController.php:51]: Same here, rather not use _POST directly

File: dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/General.xml [hunk 4]
```diff
@@ -76,6 +76,21 @@
             <MaximumValue>1000</MaximumValue>
             <ValidationMessage>Choose a value between 1 and 1000.</ValidationMessage>
         </logsize>
+        <general_log_level type="OptionField">
+            <multiple>N</multiple>
```
Comment (fichtner) [inline dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/General.xml:80]:
```suggestion
```

File: dns/bind/src/opnsense/service/conf/actions.d/actions_bind.conf [hunk 5]
```diff
@@ -40,3 +40,13 @@ parameters:
 type:script
 message:fetching DNSBLs and restart
 description: Download BIND DNSBLs and restart
+
+[check_zone]
```
Comment (fichtner) [inline dns/bind/src/opnsense/service/conf/actions.d/actions_bind.conf:44]: I'd prefer `zone.check`

File: dns/bind/src/opnsense/service/conf/actions.d/actions_bind.conf [hunk 6]
```diff
@@ -40,3 +40,13 @@ parameters:
...
+command:/usr/local/opnsense/scripts/OPNsense/Bind/checkzone.sh
+parameters: %s
+type:script_output
+
+[show_zone]
```
Comment (fichtner) [inline dns/bind/src/opnsense/service/conf/actions.d/actions_bind.conf:49]: And `zone.show`

**Comment by fichtner** [comment]
@kulikov-a so far so good... thanks! minor style updates needed please take a look :)

**Comment by kulikov-a** [comment]
@fichtner thanks! sorry for the `$_POST` and other..Sloppy on my end really (

**Comment by fichtner** [comment]
Merged, thanks! No worries. 😊

**Comment by kulikov-a** [comment]
Thanks!)

**Pull request merged**